core/src/services/gdrive/backend.rs

```diff
@@ -16,47 +16,41 @@
 // under the License.
 
 use async_trait::async_trait;
-use http::{header, Request, Response, StatusCode};
-use serde::Deserialize;
-use std::fmt::Debug;
+use http::StatusCode;
+
+use std::{collections::HashMap, fmt::Debug, sync::Arc};
+use tokio::sync::Mutex;
 
 use crate::{
     ops::{OpDelete, OpRead, OpWrite},
     raw::{
-        build_rooted_abs_path, new_request_build_error, parse_into_metadata, Accessor,
-        AccessorInfo, AsyncBody, HttpClient, IncomingAsyncBody, RpDelete, RpRead, RpWrite,
+        parse_into_metadata, Accessor, AccessorInfo, HttpClient, IncomingAsyncBody, RpDelete,
+        RpRead, RpWrite,
     },
     types::Result,
     Capability, Error, ErrorKind,
 };
 
-use super::{error::parse_error, writer::GdriveWriter};
+use super::{core::GdriveCore, error::parse_error, writer::GdriveWriter};
 
-#[derive(Clone)]
+#[derive(Clone, Debug)]
 pub struct GdriveBackend {
-    root: String,
-    access_token: String,
-    client: HttpClient,
+    core: Arc<GdriveCore>,
 }
 
 impl GdriveBackend {
     pub(crate) fn new(root: String, access_token: String, http_client: HttpClient) -> Self {
         GdriveBackend {
-            root,
-            access_token,
-            client: http_client,
+            core: Arc::new(GdriveCore {
+                root,
+                access_token,
+                client: http_client,
+                path_2_id: Arc::new(Mutex::new(HashMap::new())),
```

Comment:
How about using `path_cache`? By the way, we can use `Arc::default()` here for short.